services/identity/service.go

```diff
@@ -66,40 +65,35 @@ type Storage struct {
 	Data     *skipchain.SkipBlock
 }
 
-// NewProtocol is not used here.
-func (s *Service) NewProtocol(tn *sda.TreeNodeInstance, conf *sda.GenericConfig) (sda.ProtocolInstance, error) {
-	return nil, nil
-}
-
 /*
  * API messages
  */
 
 // CreateIdentity will register a new SkipChain and add it to our list of
 // managed identities.
-func (s *Service) CreateIdentity(si *network.ServerIdentity, ai *CreateIdentity) (network.Body, error) {
+func (s *Service) CreateIdentity(ai *CreateIdentity) (network.Body, sda.ClientError) {
 	log.Lvlf3("%s Creating new identity with config %+v", s, ai.Config)
 	ids := &Storage{
 		Latest: ai.Config,
 	}
 	log.Lvl3("Creating Root-skipchain")
-	var err error
-	ids.Root, err = s.skipchain.CreateRoster(ai.Roster, 2, 10,
+	var cerr sda.ClientError
+	ids.Root, cerr = s.skipchain.CreateRoster(ai.Roster, 2, 10,
 		skipchain.VerifyNone, nil)
-	if err != nil {
-		return nil, err
+	if cerr != nil {
+		return nil, cerr
 	}
 	log.Lvl3("Creating Data-skipchain")
-	ids.Root, ids.Data, err = s.skipchain.CreateData(ids.Root, 2, 10,
+	ids.Root, ids.Data, cerr = s.skipchain.CreateData(ids.Root, 2, 10,
 		skipchain.VerifyNone, ai.Config)
-	if err != nil {
-		return nil, err
+	if cerr != nil {
+		return nil, cerr
 	}
 
 	roster := ids.Root.Roster
 	replies, err := s.propagateIdentity(roster, &PropagateIdentity{ids}, propagateTimeout)
 	if err != nil {
-		return nil, err
+		return nil, sda.NewClientErrorCode(4100, err.Error())
```

Comment:
For this occurence and all others, is it possible to put the variable name ? I thought you had defined such constants, instead of the plain error number. Or is it too ugly ?